packages/config/src/options.js

```diff
@@ -452,17 +452,27 @@ export function getNuxtConfig (_options) {
       .map(([path, handler]) => ({ path, handler }))
   }
 
+  // App config (internal for nuxt2 at this stage)
+  const useCDN = isUrl(options.build.publicPath) && !options.dev
+  options.app = defu(options.app, {
+    basePath: options.router.base,
+    assetsPath: useCDN ? '/' : joinURL(options.router.base, options.build.publicPath),
+    cdnURL: useCDN ? options.build.publicPath : undefined
```

Comment:
is there any merit in defaulting to `'/'`?

Reply:
It is something i still doubt about but at least this ensures we always check it's existence and when it is fasly, useCDN is false

Reply:
Then I would go for null, perhaps?

Reply:
Makes sense. updated.